src/Interpreters/getCustomKeyFilterForParallelReplicas.cpp

```diff
@@ -40,55 +40,101 @@ ASTPtr getCustomKeyFilterForParallelReplica(
         return equals_function;
     }
 
-    assert(filter_type == ParallelReplicasCustomKeyFilterType::RANGE);
+    chassert(filter.filter_type == ParallelReplicasCustomKeyFilterType::RANGE);
 
     KeyDescription custom_key_description
         = KeyDescription::getKeyFromAST(custom_key_ast, columns, context);
 
     using RelativeSize = boost::rational<ASTSampleRatio::BigNum>;
 
-    RelativeSize size_of_universum = 0;
+    RelativeSize range_upper = filter.range_upper > 0 ? RelativeSize(filter.range_upper) + RelativeSize(1)
+                                                      : RelativeSize(std::numeric_limits<UInt32>::max()) + RelativeSize(1);
```

Comment:
not sure but I think it should be
```suggestion
    RelativeSize range_upper = 0;
```
I think this was mistake
```cpp
size_of_universum = RelativeSize(std::numeric_limits<UInt32>::max()) + RelativeSize(1);
```

Otherwise I'm not sure when can this trigger :thinking: 
```cpp
if (size_of_universum == RelativeSize(0))
```

Reply:
I have updated all the tests to use Uint rather than Int. Otherwise this change causes them to fail.